### PR TITLE
adding updates to deploy spire on ocp46 successfully

### PR DESCRIPTION
**Pull request by rahug20**
**Please provide enough information so that others can review your pull request:**

1. Explain the **details** for making this change. What existing problem does the pull request solve? If it resolves an existing issue, be sure to use a Github [keyword](https://help.github.com/en/articles/closing-issues-using-keywords) to automatically close it.

updating spire helm-chart and makefile to add extra steps to allow hostaccess/privilege to spire service-account user

2. What **changes to custom.yaml** are required?

N/A

3. Have you documented any additional setup steps or configurations options?

<br/><br/>

4. Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI.

Deploy OCP4.6 cluster

Update global.environment value to openshift

Run make install

Greymatter should be deployed successfully using spire

### CODE REVIEW

File: spire/Makefile [hunk 1]
```diff
@@ -7,13 +7,22 @@ include ../output.mk
...
 .PHONY: server
 server: wsa-spire
-	helm install $(NAME-SPIRE)-server server --set=global.environment=kubernetes $(WAITERSACHECK-SPIRE) -f ../global.yaml
+	helm install $(NAME-SPIRE)-server server $(WAITERSACHECK-SPIRE) -f ../global.yaml
 
 .PHONY: agent
-agent: wsa-spire
-	helm install $(NAME-SPIRE)-agent agent --set=global.environment=kubernetes $(WAITERSACHECK-SPIRE) -f ../global.yaml
+agent: wsa-spire envirionemt-check
+	@if [ "$(ENVIRONMENT)" == "openshift" ]; then \
```
Comment (chrisbsmith) [inline spire/Makefile:19]: Does this command work on both OCP 4.6 and 3.11?

**Review by chrisbsmith — CHANGES_REQUESTED** [review]
I left a comment in the makefile, but also you need to review the failed merge checks and fix the findings before this PR can be merged.

File: spire/Makefile (hunk 1 above)
Comment (rahug20) [reply spire/Makefile:19]: I believe it does. I can create OCP3.11 cluster and test it again

Comment (rahug20) [reply spire/Makefile:19]: yes this command works on both. I had tested it on minishift OCP3.11

File: spire/agent/templates/daemonset.yaml [hunk 2]
```diff
@@ -71,7 +71,7 @@ spec:
             initialDelaySeconds: 15
             periodSeconds: 60
             timeoutSeconds: 3
-          {{ if eq .Values.global.environment "openshift" }}
+          {{- if eq (default $.Values.global.environment "kubernetes") "openshift" }}
```
Comment (chrisbsmith) [inline spire/agent/templates/daemonset.yaml:74]: I don't think you want the leading `-` here do you?

File: spire/Makefile (hunk 1 above)
Comment (chrisbsmith) [reply spire/Makefile:19]: 👍

**Review by chrisbsmith — COMMENTED** [review]
One more comment on your update to the template

File: spire/agent/templates/daemonset.yaml (hunk 2 above)
Comment (rahug20) [reply spire/agent/templates/daemonset.yaml]: `{{- if eq (default $.Values.global.environment "kubernetes") "openshift" }}` was not templating `if` statement so updated the code.
should be good to go now

**Review by chrisbsmith — APPROVED** [review]
LGTM

**Pull request merged**